alf/summary/render.py

```diff
@@ -0,0 +1,658 @@
+            name="action_dist", act_dist=action_dist, action_spec=self._action_spec)
+
+        return AlgStep(
+            output=action,
+            info=dict(
+                action_img=action_img,
+                action_heatmap=action_heatmap,
+                action_dist_curve=act_dist_curve))
+"""
+
+
+class Image(object):
+    """A simple image class."""
+
+    def __init__(self, img):
+        """
+        Args:
+            img (np.ndarray): a numpy array image of shape ``[H,W]`` (gray-scale)
+                or ``[H,W,3]`` (RGB).
+        """
+        assert isinstance(img, np.ndarray), "Image must be a numpy array!"
+        shape = img.shape
+        assert (len(shape) == 2) or (len(shape) == 3 and shape[-1] == 3), (
+            "Image shape should be [H,W] or [H,W,3]!")
+        if len(shape) == 2:
+            self._img = cv2.cvtColor(img, cv2.COLOR_GRAY2RGB)
+        else:
+            self._img = img
+
+    @property
+    def shape(self):
+        """Return the shape of the image."""
+        return self._img.shape
+
+    @property
+    def data(self):
+        """Return the image numpy array which is always RGB."""
+        return self._img
+
+    def resize(self, height=None, width=None):
+        """Resize the image in-place given the desired width and/or height.
+
+        Args:
+            height (int): the desired output image height. If ``None``, this will
+                be scaled to keep the original aspect ratio if ``width`` is provided.
+            width (int): the desired output image width. If ``None``, this will
+                be scaled to keep the original aspect ratio if ``height`` is
+                provided.
+        """
+        if width is not None and height is not None:
+            self._img = cv2.resize(self._img, dsize=(width, height))
+            return
+        if width is not None:
+            scale = float(width) / self._img.shape[1]
+        elif height is not None:
+            scale = float(height) / self._img.shape[0]
+        else:
+            raise ValueError('At least width or height should be provided.')
+        self._img = cv2.resize(
+            self._img,
+            dsize=(0, 0),
+            fx=scale,
+            fy=scale,
+            interpolation=cv2.INTER_LINEAR)
+
+    @classmethod
+    def from_pyplot_fig(cls, fig, dpi=200):
+        """Generate an ``Image`` instance from a pyplot figure instance.
+
+        Args:
+            fig (pyplot.figure): a pyplot figure instance
+            dpi (int): resolution of the generated image
+
+        Returns:
+            Image:
+        """
+        buf = io.BytesIO()
+        fig.savefig(buf, format="png", dpi=dpi, bbox_inches="tight")
+        buf.seek(0)
+        img_arr = np.frombuffer(buf.getvalue(), dtype=np.uint8)
+        buf.close()
+        img = cv2.imdecode(img_arr, 1)
+        img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
+        return cls(img)
+
+    @classmethod
+    def from_image_nest(cls, imgs):
```

Comment:
```from_image_nest```: can change the name of function to be more informative, e.g. ```pack_image_nest```?

Reply:
This is a class method. I think it's quite clear Image.from_image_nest.